.github/workflows/auto-close-github-discussions.yml

```diff
@@ -0,0 +1,304 @@
+name: Auto Close Stale Discussions
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Runs daily at midnight UTC
+  workflow_dispatch: # Enables manual run from the Actions tab
+
+permissions:
+  contents: read
+  discussions: write
+
+jobs:
+  close-discussions:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
```

Comment:
Why checkout the repository?

Reply:
Previously it wasn't needed. Now, as you mentioned , this will make sense as we have moved the script to a separate file.

Thanks for the suggestion!